.github/workflows/matrix-tests.yaml

```diff
@@ -6,14 +6,29 @@ on:
   workflow_dispatch:
 
 jobs:
-  main:
+  set-matrix:
     runs-on: ubuntu-latest
-    name: Run the interface tests on all registered charms
+    outputs:
+      matrix_values: ${{ steps.set-matrix.outputs.matrix_values }}
     steps:
       - name: Checkout
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
         with:
           fetch-depth: 1
+      - name: Set matrix values
+        id: set-matrix
+        run: |
+          interfaces=[$(find interfaces -mindepth 1 -maxdepth 1 -type d -not -name "__template__" -printf '"%f",')]
```

Comment:
This is a bit awkward, trying to generate JSON. I'm also surprised it's works with `fromJSON` as with the trailing comma it's not actually correct JSON (maybe `fromJSON` is more permissive?).

Is `jq` available? If so, maybe this is better than trying to build JSON by hand?

```
find interfaces -mindepth 1 -maxdepth 1 -type d -not -name "__template__" | jq --raw-input . | jq --slurp .
```

Also, does this work for things like [`ingress`](https://github.com/canonical/charm-relation-interfaces/tree/main/interfaces/ingress), which have multiple versions, v1 and v2 in that case?

Reply:
I tried, `jq` is available, and I changed it to `jq`. I still kept the `-printf '%f\n'` part for the find command so that it strips the parent dir name easily.

Interfaces with multiple versions: yes they should still work, if any version fails, the job for that interface will fail.